### ISSUE TITLE
[bug] Meson build helper overrides profile env variables when compiling with MSVC+ninja

### ISSUE THREAD

**Issue by agurtovoy** [conan-io/conan#5999]
When building with Conan+Meson+ninja on Windows, conan ends up shadowing PATH env variable's value specified in the corresponding conan profile with the value coming from "vcvars". For example, given this profile:

```
SQUISHCOCO="C:\Program Files\squishcoco"

[env]
PATH=[$SQUISHCOCO\visualstudio_x64;$SQUISHCOCO]
COVERAGESCANNER_ARGS=--cs-on --cs-mcdc --cs-mcc
```

one would expect the PATH variable at the time of the Meson invocation to start with `C:\Program Files\squishcoco\visualstudio_x64;C:\Program Files\squishcoco;`. In practice, it ends up looking more like this: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\bin\HostX64\x64;<more MSVC paths...>;C:\Program Files\squishcoco\visualstudio_x64;C:\Program Files\squishcoco;<more paths...>`

**Some context**: [Squish Coco](https://doc.froglogic.com/squish-coco/latest/) is a commercial, cross-platform C/C++ code coverage tool often used in heavily-regulated software industries (medical devices, avionics, etc.). To instrument your code, you have to configure your `PATH` to point to Squish Coco's compiler/librarian/linker wrappers that mimic/redirect to the native toolchain. This issue prevents us from being able to use Conan when compiling our projects under Squish Coco.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: Visual Studio 2017
  * Conan version: 1.20.0-dev
  * Python version: 3.8.0

**Comment by nicememory** [comment]
I don't think it's a issue with Meson only, it's a more generic issue about env precedence. Invoking order and side effects must be considered. Consistency is important, but when things go wrong, there should be a way to make it work without ugly hacks.

**Comment by agurtovoy** [comment]
I realized that the same applies to at least the CMake helper when working on #6000, just wasn't ready to tackle a broader fix without some guidance/feedback.

**Comment by agurtovoy** [comment]
With the exception of the hard-coded name of the build helper class, this test could be used to reproduce/verify the absence of the issue elsewhere: https://github.com/conan-io/conan/pull/6000/files#diff-9201a549e904721ccf01e164da8a7fc9R203

**Comment by agurtovoy** [comment]
Another example where this is causing issues: latest versions of Visual Studio ship with their own CMake and Ninja binaries, and running `vcvars` after applying the environment from Conan profiles gives preference to those binaries over what's specified in the profile.

**Comment by memsharded** [comment]
Merged, will be released in 1.22

**Comment by mgrandsire** [comment]
Hi @agurtovoy, 

I'm wondering if you could provide your complete profile for squish coco. I'm trying to add it but I'm facing to many errors.
I try to call "csg++" and "csgcc" and also to set "--cs-on --cs-mcdc" option 

Thanks
Mickael


PS: I don't know if it is authorized to write new comments in a closed ticket and where the topics is slightly different because nothing appears in contributing guideline but this request could be helpful for many users. I'll create a new topics if needed. Sorry in advance.
